src/stan-users-guide/latent-discrete.Rmd

```diff
@@ -1037,3 +1037,89 @@ the probability mass function of $z_i$ in the posterior.   These may
+
+Introduction {-}
+
+This section describes in more detail the mathematics of statistical inference using the output of marginalized Stan models, such as those presented in the last three sections. It provides a mathematical explanation of why and how certain manipulations of Stan's output produce valid summaries of the posterior distribution when discrete parameters have been marginalized out of a statistical model. Ultimately, however, fully understanding the mathematics in this section is *not* necessary to fit models with discrete parameters using Stan.  
+
+Throughout, the model under consideration consists of both continuous parameters, $\Theta$, and discrete parameters, $Z$. It is also assumed that $Z$ can only take finitely many values, as is the case for all the models described in this chapter of the User's Guide. To simplify notation, any conditioning on data is suppressed in this section, except where specified. As with all Bayesian analyses, however, all inferences using models with marginalized parameters are made conditional on the observed data. 
+
+Estimating expectations {-}
+
+When performing Bayesian inference, interest often centres on estimating some (constant) low-dimensional summary statistics of the posterior distribution. Mathematically, we are interested in estimating $\mu$, say, where $\mu = \mathbb{E}(g(\Theta, Z))$ and $g(\cdot)$ is an arbitrary function. An example of such a quantity is $\mathbb{E}(\Theta)$, the posterior mean of the continuous parameters, where we would take $g(\theta, z) = \theta$. To estimate $\mu$ the most common approach is to draw a series of values, at least approximately, from the posterior distribution of the parameters of interest. The numerical values of these draws can then be used to calculate the quantities of interest. Often, this process of calculation is trivial, but more care is required when working with marginalized posteriors as we describe in this section.
+
+If both $\Theta$ and $Z$ were continuous, Stan could be used to sample $M$ draws from the joint posterior $p_{\Theta, Z}(\theta, z)$ and then estimate $\mu$ with
+$$   
+\hat{\mu} = \frac{1}{M} \sum_{i = 1}^M {g(\theta_i, z_i)}.
+$$
+Given $Z$ is discrete, however, Stan cannot be used to sample from the joint posterior because Stan's NUTS algorithm cannot compute the derivatives it needs for the discrete parameters. Instead, as outlined in the previous sections describing specific models, the user can first marginalize out $Z$ from the joint posterior to give the marginalized posterior $p_\Theta(\theta)$. This marginalized posterior can then be implemented in Stan as usual, and Stan will give samples $\{\theta_i\}_{i = 1}^M$ from the marginalized posterior. 
+
+Using only the samples $\{\theta_i\}_{i = 1}^M$, how can we estimate $\mathbb{E}(g(\Theta, Z))$? We can use a conditional estimator. That is, for each function $g$ of interest we can compute
```

Comment:
Update notation for draws throughout.   

We've also tried to stick with the standard statistical notion of a sample consisting of a bunch of draws, but lots of people conflate "sample" and "draw", so fixing the language isn't critical.